src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1138,6 +1138,36 @@ private AuthenticationResult acquireTokenLocalCall(final CallbackHandler callbac
     private boolean promptUser(PromptBehavior prompt) {
         return prompt == PromptBehavior.Always || prompt == PromptBehavior.REFRESH_SESSION;
     }
+    
+    
+    //checkRedictUri() checks the redirectUri before sending the request
+    //if the redirectUri from the client does not match the valid redirectUri
+    // the client app would not jump to the login page
+    //redirectUri format %PREFIX://%PACKAGE_NAME/%SIGNATURE   
+    private boolean checkRedictUri(CallbackHandler callbackHandle, final AuthenticationRequest request) throws UnsupportedEncodingException{
+    	
+    	String errMsg = new String();    	
+    	String inputUri = request.getRedirectUri();
+    	PackageHelper helper = new PackageHelper(mContext);
+    	
+    	if(inputUri == null){
```

Comment:
you also need to verify if inputUri is empty. Use StringExtensions.isNullorBlank